### PR TITLE
Fix app separation

### PR DESCRIPTION
**Pull request by muffinresearch**
Fixes: #247
Fixes: #216 

This removes the core app from being able to be run so you now need to run/develop each app separately. 

That's probably not a bad thing in the long run as crosstalk is undesirable. Good testing should ensure that core changes don't negatively impact each app.

I added a clean step to the dev server runs as switching between the app confuses things because a webpack-assets.json pre-exists. This might explain some of the first run weirdness we've been seeing on occasion.

### CODE REVIEW

**Comment by muffinresearch** [comment]
Oops forgot about tests. Let me fix that...

File: tests/server/TestDiscoViews.js [hunk 1]
```diff
@@ -20,9 +20,9 @@ describe('GET requests', () => {
     webpackIsomorphicTools.undo();
   });
 
-  for (const path of ['search', 'disco']) {
+  for (const path of ['/']) {
```
Comment (mstriemer) [inline tests/server/TestDiscoViews.js]: This loop can go away.

File: tests/server/TestSearchViews.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    .then((server) => {
+      app = server;
+      done();
+    }));
+
+  after(() => {
+    webpackIsomorphicTools.undo();
+  });
+
+  for (const path of ['/search']) {
```
Comment (mstriemer) [inline tests/server/TestSearchViews.js]: Looks like this could go too.

File: tests/server/TestDiscoViews.js (hunk 1 above)
Comment (muffinresearch) [inline tests/server/TestDiscoViews.js]: Yeah I wasn't sure if we might want to add to it later. But certainly I can clean it up as we can easily add it back as needed.

**Comment by muffinresearch** [comment]
@mstriemer let me know if there's any other nits. I'm landing this so I can move ahead with the disco pane components.

File: src/config/index.js [hunk 3]
```diff
@@ -62,11 +63,18 @@ const config = new Map();
...
-  const APP_NAMES = ['search', 'disco'];
-  if (APP_NAMES.indexOf(process.env.APP_NAME) < 0) {
-    config.set('currentApp', 'core');
+  // This is the list of apps allowed to run.
+  const validAppNames = ['search', 'disco'];
+  config.set('validAppNames', validAppNames);
+
+  // Create a list of apps to build targets for.
+  const appName = process.env.APP_NAME;
+  if (validAppNames.indexOf(appName) > -1) {
```
Comment (mstriemer) [inline src/config/index.js]: We should probably use `validAppNames.includes(appName)` instead of using `indexOf()` now.

Comment (muffinresearch) [inline src/config/index.js]: Yep good call - fixed in https://github.com/mozilla/addons-frontend/commit/29c65944414aa15ed75742b7142935e4be54826d

Comment (muffinresearch) [inline src/config/index.js]: I had to revert that change as it's not in node. Will look at getting the polyfill working at some point but for quickness I reverted it.

**Pull request merged**